src/main/java/org/openlmis/buq/web/buq/BottomUpQuantificationController.java

```diff
@@ -0,0 +1,216 @@
+    }
+
+    BottomUpQuantification updatedBottomUpQuantification = bottomUpQuantificationService
+        .save(bottomUpQuantificationDto, bottomUpQuantificationId);
+
+    return buildDto(updatedBottomUpQuantification);
+  }
+
+
+  /**
+   * Retrieves audit information related to the specified bottom-up quantification.
+   *
+   * @param author The author of the changes which should be returned.
+   *               If null or empty, changes are returned regardless of author.
+   * @param changedPropertyName The name of the property about which changes should be returned.
+   *               If null or empty, changes associated with any and all properties are returned.
+   * @param page A Pageable object that allows client to optionally add "page" (page number)
+   *             and "size" (page size) query parameters to the request.
+   */
+  @GetMapping(value = "/{id}/auditLog")
+  @ResponseStatus(HttpStatus.OK)
+  @ResponseBody
+  public ResponseEntity<String> getBottomUpQuantificationAuditLog(@PathVariable("id") UUID id,
+      @RequestParam(name = "author", required = false, defaultValue = "") String author,
+      @RequestParam(name = "changedPropertyName", required = false, defaultValue = "")
+      String changedPropertyName, Pageable page) {
+
+    // Return a 404 if the specified instance can't be found
+    if (!bottomUpQuantificationRepository.existsById(id)) {
+      throw new NotFoundException(MessageKeys.ERROR_BOTTOM_UP_QUANTIFICATION_NOT_FOUND);
+    }
+
+    return getAuditLogResponse(BottomUpQuantification.class, id, author, changedPropertyName,
+        page);
+  }
+
+  private BottomUpQuantificationDto buildDto(BottomUpQuantification bottomUpQuantification) {
+    BottomUpQuantificationDto dto = new BottomUpQuantificationDto();
+    bottomUpQuantification.export(dto);
+
+    List<BottomUpQuantificationLineItem> bottomUpQuantificationLineItems =
+        bottomUpQuantification.getBottomUpQuantificationLineItems();
+    List<BottomUpQuantificationLineItemDto> lineItemDtoList = bottomUpQuantificationLineItems
+        .stream()
+        .map(lineItem -> {
+          BottomUpQuantificationLineItemDto lineItemDto = new BottomUpQuantificationLineItemDto();
+          lineItem.export(lineItemDto);
+
+          return lineItemDto;
+        })
+        .collect(Collectors.toList());
+    dto.setBottomUpQuantificationLineItems(lineItemDtoList);
+
+    List<BottomUpQuantificationStatusChange> statusChanges =
+        bottomUpQuantification.getStatusChanges();
+    List<BottomUpQuantificationStatusChangeDto> statusChangeDtos = statusChanges
+        .stream()
+        .map(statusChange -> {
+          BottomUpQuantificationStatusChangeDto statusChangeDto =
+              new BottomUpQuantificationStatusChangeDto();
+          statusChange.export(statusChangeDto);
+
+          return statusChangeDto;
+        })
+        .collect(Collectors.toList());
+    dto.setStatusChanges(statusChangeDtos);
+
+    return dto;
```

Comment:
shouldn't it be a service method?

Reply:
I created a separate component, BottomUpQuantificationDtoBuilder, responsible for this functionality and extract this method to it.